### DISCUSSION TITLE
Getting 2x2 working for the baroclinic wave on the sphere

### DISCUSSION THREAD

**Discussion by Witt-D** [firedrakeproject/gusto#546] [Design discussion]
Discussion on the adjustments to the solver and solver parameters needed for the 2x2 ratio of inner / outer loops in the SIQN solver

**Comment by Witt-D** [comment]
A timeline of what has happened so far. 

When I first tried the 2x2 solver options for the Baroclinic Wave, It would cause the simulation to crash after just a handful of timesteps.
I would consider the 2x2 to be successfully implemented once it reaches 15 days of simulation, which matches the 4x1 solver options.

To Try and fix this, the following steps have occurred: 

- I moved the Coriolis force to be solved as part of the linear solver. This is now on Main and improved the simulation to run up to 6 days of integration
- I passed `tau_values={'rho':1.0, 'theta':1.0}` Into the linear solver. This improved simulation run to 10 days

Next I will try an experimental branch (as of 22/08/24) in which the reference profile is updated more frequently as opposed to just being a balanced state at the start

**Comment by Witt-D** [comment]
Update: I ran the baroclinic wave with the `tau_values={'rho':1.0, 'theta':1.0}` options into the linear solver on a branch that updated the references profiles as the integration went on. Unfortunately, the simulation still crashed on day 10 of the simulation.
The final state of the simulation on day 10 looks sensible, however.

I am now trying slightly adjusting the centering value `alpha=0.55` to see if that helps

**Comment by tommbendall** [comment]
A very belated comment on this, but we believe we fixed this in #645 by ensuring the `accelerator` (which zeros residuals for density and potential temperature for inner > 0) is correct.
